src/java.base/share/classes/java/lang/reflect/ReflectAccess.java

```diff
@@ -127,4 +127,9 @@ public boolean isTrustedFinalField(Field f) {
     {
         return ctor.newInstanceWithCaller(args, true, caller);
     }
+
+    public Object invokeDefault(Object proxy, Method method, Object[] args, Class<?> caller)
+            throws Throwable {
+        return Proxy.invokeDefault(proxy, method, args, caller);
+    }
```

Comment:
What about other, non-jdk code that wishes to implement a similar thing - make a proxy for an arbitrary interface and handle default methods by invoking them?

Reply:
To invoke the default method, the caller will need access to the declaring interface of the default method (via bytecode invocation or reflection).   The bug I had was because java.base does not have access to the interface even though the caller has.

Reply:
> To invoke the default method, the caller will need access to the declaring interface of the default method (via bytecode invocation or reflection). The bug I had was because java.base (the module of the invocation handler) does not have access to the interface even though the caller has.

So invokeDefault is moved to Proxy with an optional access check, and you've made it callable from MHProxies by way of JLRA. I think that is okay.